scripts/rv_expression_association/prepare_inputs.py

```diff
@@ -0,0 +1,211 @@
+        'limix==3.0.4',
+        'pandas_plink==2.2.9',
+        'xarray==0.20.2',
+    ],
+    check=True,
+)
+
+import xarray as xr  # pylint: disable=wrong-import-position, import-error
+from pandas_plink import (read_plink1_bin)  # pylint: disable=wrong-import-position, import-error
+from limix.qc import (quantile_gaussianize)  # pylint: disable=wrong-import-position, import-error
+
+# use logging to print statements, display at info level
+logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.INFO)
+
+
+@click.command()
+@click.option('--cell-type', required=True)  # 'B_naive'
+@click.option('--gene-name', required=True)  # 'VPREB3'
+@click.option(
+    '--sample-mapping-file', required=True
+)  # 'scrna-seq/grch38_association_files/OneK1K_CPG_IDs.tsv'
+@click.option(
+    '--genotype-file-bed', required=True
+)  # 'v0/plink_files/vpreb3_rare_promoter.bed'
+@click.option(
+    '--genotype-file-bim', required=True
+)  # 'v0/plink_files/vpreb3_rare_promoter.bim'
+@click.option(
+    '--genotype-file-fam', required=True
+)  # 'v0/plink_files/vpreb3_rare_promoter.fam'
+@click.option(
+    '--phenotype-file', required=True
+)  # 'scrna-seq/grch38_association_files/expression_files/B_naive_expression.tsv'
+@click.option('--kinship-file', required=True)  # 'v0/skat/grm_wide.csv'
+def prepare_inputs(  # pylint: disable=missing-function-docstring, too-many-locals
+    cell_type: str,
+    gene_name: str,
+    sample_mapping_file: str,
+    genotype_file_bed: str,
+    genotype_file_bim: str,
+    genotype_file_fam: str,
+    phenotype_file: str,
+    kinship_file: str,
+):
+
+    expression_filename = AnyPath(output_path(f'{gene_name}_{cell_type}.csv'))
+    genotype_filename = AnyPath(output_path(f'{gene_name}_rare_regulatory.csv'))
+    kinship_filename = AnyPath(output_path('kinship_common_samples.csv'))
+
+    # region PHENOTYPE_FILE
+
+    phenotype = pd.read_csv(phenotype_file, sep='\t', index_col=0)
+
+    phenotype = xr.DataArray(
+        phenotype.values,
+        dims=['sample', 'gene'],
+        coords={'sample': phenotype.index.values, 'gene': phenotype.columns.values},
+    )
+
+    # endregion PHENOTYPE_FILE
+
+    # region KINSHIP_FILE
+
+    # read in GRM (genotype relationship matrix; kinship matrix)
+    kinship = pd.read_csv(kinship_file, index_col=0)
+    kinship.index = kinship.index.astype('str')
+    assert all(kinship.columns == kinship.index)  # symmetric matrix, donors x donors
+
+    kinship = xr.DataArray(
+        kinship.values,
+        dims=['sample_0', 'sample_1'],
+        coords={'sample_0': kinship.columns, 'sample_1': kinship.index},
+    )
+    kinship = kinship.sortby('sample_0').sortby('sample_1')
+
+    # endregion KINSHIP_FILE
+
+    # region GENOTYPE_FILE
+
+    # read in genotype file (plink format)
+    # bed
+    with to_path(genotype_file_bed).open('rb') as handle:
+        data = handle.readlines()
+    with open('temp.bed', 'wb') as handle:
+        handle.writelines(data)
```

Comment:
```suggestion
    with to_path(genotype_file_bed).open('rb') as read_handle:
        with open('temp.bed', 'wb') as write_handle:
            write_handle.writelines(read_handle.readlines())
```

This is a little fiddly, but doesn't require any intermediate object to hold the file content

Reply:
ok, as long as it works! :)

Reply:
Why not just copy this to disk?

```python
to_path(genotype_file_bed).copy('temp.bed')

Reply:
Looks great, didn't know that method existed
https://cloudpathlib.drivendata.org/~latest/api-reference/cloudpath/#cloudpathlib.cloudpath.CloudPath.copy

Reply:
perfect thank you!!